momepy/preprocessing.py

```
@@ -1101,3 +1105,464 @@ def roundabout_simplification(
+        f"origin_cluster != destination_cluster or (('{edge_to_col}' == '{edge_from_col}') and origin_cluster >= 0)"
+    )
+
+    # Determine the new point geometries (when exists):
+    edges_tosimplify_gdf = edges_tosimplify_gdf.assign(
+        new_origin_pt=edges_tosimplify_gdf.origin_cluster.map(
+            cluster_gdf.geometry, None
+        )
+    )
+    edges_tosimplify_gdf = edges_tosimplify_gdf.assign(
+        new_destination_pt=edges_tosimplify_gdf.destination_cluster.map(
+            cluster_gdf.geometry, None
+        )
+    )
+
+    # Determine the new geometry according to the simplification method:
+    if method == "extend":
+        edges_simplified_geometries = edges_tosimplify_gdf.apply(
+            lambda edge: _extension_simplification(
+                edge.geometry, edge.new_origin_pt, edge.new_destination_pt
+            ),
+            axis=1,
+        )
+        edges_simplified_gdf = edges_tosimplify_gdf.assign(
+            new_geometry=edges_simplified_geometries
+        )
+    elif method == "euclidean":
+        edges_simplified_geometries = edges_tosimplify_gdf.apply(
+            lambda edge: _euclidean_simplification(
+                edge.geometry, edge.new_origin_pt, edge.new_destination_pt
+            ),
+            axis=1,
+        )
+        edges_simplified_gdf = edges_tosimplify_gdf.assign(
+            new_geometry=edges_simplified_geometries
+        )
+    elif method == "spider":
+        edges_simplified_geometries = edges_tosimplify_gdf.apply(
+            lambda edge: _spider_simplification(
+                edge.geometry, edge.new_origin_pt, edge.new_destination_pt, buffer
+            ),
+            axis=1,
+        )
+        edges_simplified_gdf = edges_tosimplify_gdf.assign(
+            new_geometry=edges_simplified_geometries
+        )
+    else:
+        warnings.warn(
+            f"Simplification method '{method}' not recognized. Using spider-web simplification."
+        )
```

I don't think this is a good behaviour. We should raise an error if the method is not implemented.

---

threw a `ValueError` in that case